packages/shared/src/resource-archiver/index.ts

```diff
@@ -110,40 +110,17 @@ export class Watcher {
 
     // Pausing a response stage with a response
     if (responseStatusCode) {
-      if ([301, 302, 307, 308].includes(responseStatusCode)) {
-        await this.clientSend(request, 'Fetch.continueRequest', {
+      await this.handleSuccessfulResponse(
+        {
+          request,
           requestId,
-          interceptResponse: true,
-        });
-        return;
-      }
-
-      const result = await this.clientSend(request, 'Fetch.getResponseBody', {
-        requestId,
-      });
-      // Something has gone wrong and will be logged above
-      if (result === null) {
-        return;
-      }
-      const { body, base64Encoded } = result;
-
-      // If the Content-Type header is present, let's capture it.
-      const contentTypeHeader: Protocol.Fetch.HeaderEntry = responseHeaders.find(
-        ({ name }) => name.toLowerCase() === 'content-type'
+          responseStatusCode,
+          responseStatusText,
+          responseHeaders,
+        },
```

Comment:
Codacy was not happy with the complexity level (`14`) in this file, so I extracted the response-is-successful-so-we'll-archive stuff into its own method. Aside from extracting, I didn't change any behavior there.